test/disk_uploader_test.go

```diff
@@ -0,0 +1,112 @@
+package test
+
+import (
+	"context"
+	"os"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclientcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+	kubevirtcliv1 "kubevirt.io/client-go/kubecli"
+	cdiv1beta1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/tests/libdv"
+
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/constants"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/framework"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/runner"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/testconfigs"
+	"github.com/kubevirt/kubevirt-tekton-tasks/test/vm"
+)
+
+var _ = Describe("Run disk-uploader", func() {
+	f := framework.NewFramework()
+
+	BeforeEach(func() {
+		if f.TestOptions.SkipDiskUploaderTests {
+			Skip("skipDiskUploaderTests is set to true, skipping tests")
+		}
+	})
+
+	It("Extracts disk from VM and upload to container registry", func() {
+		imageDestination := os.Getenv("REGISTRY_IMAGE_DESTINATION")
+		Expect(imageDestination).ToNot(BeEmpty())
+
+		registryCredentials, err := createRegistryCredentials(f.CoreV1Client, f.DeployNamespace)
+		Expect(err).ToNot(HaveOccurred())
+
+		cirrosVm, err := createCirrosVM(f.KubevirtClient, f.DeployNamespace)
+		Expect(err).ToNot(HaveOccurred())
+
+		_, err = vm.WaitForVM(f.KubevirtClient, f.DeployNamespace, cirrosVm.Name, "", constants.Timeouts.WaitForVMStart.Duration, false)
+		Expect(err).ToNot(HaveOccurred())
```

Comment:
Move this to BeforeEach?

Reply:
Why? this is a specific initialization of a VM for this test case, we don't have more test cases that can benefit from it at the moment. Also, it would require to introduce variables outside this scope.